crates/autopilot/src/domain/events/mod.rs

```diff
@@ -0,0 +1,205 @@
+            .await
+            .map_err(Error::Persistence)?;
+        self.persistence
+            .delete_trade_events(tx, *range.start())
+            .await
+            .map_err(Error::Persistence)?;
+        self.persistence
+            .delete_cancellation_events(tx, *range.start())
+            .await
+            .map_err(Error::Persistence)?;
+        self.persistence
+            .delete_presignature_events(tx, *range.start())
+            .await
+            .map_err(Error::Persistence)
+    }
+
+    async fn insert(
+        &self,
+        tx: &mut Transaction,
+        events: Vec<boundary::Event<boundary::events::GPv2Contract>>,
+    ) -> Result<(), Error> {
+        let mut settlements = Vec::new();
+        let mut trades = Vec::new();
+        let mut cancellations = Vec::new();
+        let mut presignatures = Vec::new();
+
+        for event in events {
+            let metadata = match event.meta {
+                Some(meta) => meta,
+                None => {
+                    tracing::warn!(?event, "GPv2Contract event without meta data");
+                    continue;
+                }
+            };
+            match event.data {
+                boundary::events::GPv2Contract::Settlement(event) => {
+                    settlements.push(settlement::Settlement {
+                        block_number: metadata.block_number,
+                        log_index: metadata.log_index,
+                        tx_hash: metadata.transaction_hash,
+                        solver: event.solver,
+                    });
+                }
+                boundary::events::GPv2Contract::Trade(event) => {
+                    trades.push(settlement::Trade {
+                        block_number: metadata.block_number,
+                        log_index: metadata.log_index,
+                        uid: uid(&event.order_uid.0)?,
+                        sell_amount_including_fee: event.sell_amount,
+                        buy_amount: event.buy_amount,
+                        fee_amount: event.fee_amount,
+                    });
+                }
+                boundary::events::GPv2Contract::OrderInvalidated(event) => {
+                    cancellations.push(settlement::Cancellation {
+                        block_number: metadata.block_number,
+                        log_index: metadata.log_index,
+                        uid: uid(&event.order_uid.0)?,
+                    });
+                }
+                boundary::events::GPv2Contract::PreSignature(event) => {
+                    presignatures.push(settlement::PreSignature {
+                        block_number: metadata.block_number,
+                        log_index: metadata.log_index,
+                        uid: uid(&event.order_uid.0)?,
+                        owner: event.owner,
+                        signed: event.signed,
+                    });
+                }
+                boundary::events::GPv2Contract::Interaction(_) => {}
```

Comment:
This might be useful information to have. Is it intended to store this at some point?

Reply:
🤷 at some point if someone needs it maybe. This PR is focussing on porting the existing logic without any changes.